src/ripple/rpc/impl/Role.cpp

```diff
@@ -137,10 +131,20 @@ forwardedFor(http_request_type const& request)
             return {};
 
         found += forStr.size();
-        auto pos{it->value().find(';', forStr.size())};
+        auto pos = std::string(found).find(';');
```

Comment:
can you do this without the temporary string ?(e.g. using the former string view methods, but perhaps correcting the start offset). I don't this is safe and in-fact the asan build is flagging it.

Reply:
fixed